src/Nethermind/Nethermind.Blockchain.Test/FullPruning/FullPrunerTests.cs

```diff
@@ -221,15 +221,18 @@ public async Task<bool> WaitForPruning()
 
             public async Task<bool> WaitForPruningEnd(TestFullPruningDb.TestPruningContext context)
             {
-                await Task.Yield();
-                await context.WaitForFinish.WaitOneAsync(TimeSpan.FromMilliseconds(Timeout.MaxWaitTime * 5), CancellationToken.None);
+                while (!await context.WaitForFinish.WaitOneAsync(TimeSpan.FromMilliseconds(1), CancellationToken.None))
+                {
+                    AddBlocks(1);
+                }
                 AddBlocks(1);
                 return await context.DisposeEvent.WaitOneAsync(TimeSpan.FromMilliseconds(Timeout.MaxWaitTime * 5), CancellationToken.None);
             }
 
             public TestFullPruningDb.TestPruningContext WaitForPruningStart()
             {
                 PruningTrigger.Prune += Raise.Event<EventHandler<PruningTriggerEventArgs>>();
+                Thread.Sleep(10);
```

Comment:
Can we avoid Thread.Sleep in the tests? Expose some task to await in tests or something?

Reply:
Updated to reduce sleep. Can't find a way to eliminate them.